output/sarif_test.go

```diff
@@ -0,0 +1,1448 @@
+package output
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	// Test file and namespace
+	testFileName  = "examples/kubernetes/service.yaml"
+	testNamespace = "namespace"
+
+	// Test messages
+	testFailureMsg = "first failure"
+	testWarningMsg = "first warning"
+	testSecondWarn = "second warning"
+
+	// Test metadata
+	testFailureDesc = "A detailed description of the failure"
+	testFailureURL  = "https://example.com/docs"
+	testFailureHelp = "How to fix this failure"
+	testWarnDesc    = "A detailed description of the warning"
+	testWarnURL     = "https://example.com/warnings"
+	testWarnHelp    = "How to fix this warning"
+
+	// Test policy metadata
+	testPackage  = "security.container"
+	testRuleName = "no_root_user"
+)
+
+func TestSARIF(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []CheckResult
+		validate func(t *testing.T, output string)
```

Comment:
We should not need a custom function for each test case. Since the output of SARIF is JSON, can we instead provide an expected JSON message and diff them?

Reply:
Good idea